### PR TITLE
[Excel] (Custom Functions) Further explain custom function limitations

### PR DESCRIPTION
**Pull request by alison-mk**
Expanding description in response to customer issue #3506.

### CODE REVIEW

File: docs/excel/call-excel-apis-from-custom-function.md [hunk 1]
```diff
@@ -43,18 +43,16 @@ async function getRangeValue(address) {
 
 ## Limitations of calling Excel JavaScript APIs through a custom function
 
-Don't call Excel JavaScript APIs from a custom function that change the environment of Excel. This means your custom functions should not do any of the following:
+A custom functions add-in has the ability to call Excel JavaScript APIs, but you should be cautious about which Excel JavaScript APIs your add-in calls. Don't call Excel JavaScript APIs from a custom function that change cells outside of the cell running the custom function. Changing other cells or the Excel environment can result in poor performance, time outs, and infinite loops in the Excel application. This means your custom functions shouldn't do any of the following.
```
Comment (AlexJerabek) [inline docs/excel/call-excel-apis-from-custom-function.md:46]: Suggestions for conciseness.
```suggestion
A custom functions add-in can call Excel JavaScript APIs, but you should be cautious about which APIs it calls. Don't call Excel JavaScript APIs from a custom function that change cells outside of the cell running the custom function. Changing other cells or the Excel environment can result in poor performance, time outs, and infinite loops in the Excel application. This means your custom functions shouldn't do any of the following.
```

File: docs/excel/call-excel-apis-from-custom-function.md [hunk 2]
```diff
@@ -43,18 +43,16 @@ async function getRangeValue(address) {
...
-- Change any of the environment options, such as calculation mode or screen views.
 - Add names to a workbook.
-- Set properties or execute most methods.
+- Set properties.
+- Change any of the Excel environment options, such as calculation mode or screen views.
 
-Changing Excel can result in poor performance, time outs, and infinite loops. Custom function calculations shouldn't run while an Excel recalculation is taking place as it will result in unpredictable results.
-
-Instead, make changes to Excel from the context of a ribbon button, or task pane.
+Your custom functions add-in can retrieve information from cells outside the cell running the custom function, but it shouldn't perform write operations to other cells. Instead, make changes to other cells or to the Excel environment from the context of a ribbon button or a task pane. In addition, custom function calculations shouldn't run while an Excel recalculation is taking place, as this scenario creates unpredictable results.
```
Comment (AlexJerabek) [inline docs/excel/call-excel-apis-from-custom-function.md:55]: I like using the read/write terminology here to reinforce the underlying actions. Also, conciseness suggestion.
```suggestion
Your custom functions add-in can read information from cells outside the cell running the custom function, but it shouldn't perform write operations to other cells. Instead, make those changes from the context of a ribbon button or a task pane. In addition, custom function calculations shouldn't run while an Excel recalculation is taking place, as this scenario creates unpredictable results.
```

**Review by AlexJerabek — APPROVED** [review]
Thanks for making this change. Looks good! Just a couple suggestions to make the writing a little tighter.

File: docs/excel/call-excel-apis-from-custom-function.md (hunk 2 above)
Comment (alison-mk) [reply docs/excel/call-excel-apis-from-custom-function.md:55]: Thanks for the review and suggestions! I'll take the read/write half of this one, but retain the "changes to other cells or to the Excel environment" phrase to reinforce that the instruction to use a task pane or ribbon button applies to both changes to other cells or to Excel environment.

**Pull request merged**